### PR TITLE
add built envronment clean shell script and cmake install script && updat .travis.yml to configure travis-ci && modify build.sh

### PR DESCRIPTION
**Pull request by cyjseagull**
1. add `clean_cache.sh` && `install_cmake.sh`: `clean_cache.sh` is used to clean built packages; `install_cmake.sh` used to install higher version cmake
2. update `.travis.yml` : configure travis-ci
3. modify `build.sh` : to make building procedure of travis-ci passed
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/wheatli%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/23206367%3Fv%3D4%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23issuecomment-414948040%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A05%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/23206367%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wheatli%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222018-08-22T09%3A10%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/23206367%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wheatli%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%203ec01471b0598b5af99fa69e1767f979db7837aa%20tools/clean_cache.sh%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211841827%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22owner%20should%20be%20fisco-bcos%22%2C%20%22created_at%22%3A%20%222018-08-22T06%3A33%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/23206367%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wheatli%22%7D%7D%2C%20%7B%22body%22%3A%20%22owner%20is%20modified%20to%20fisco-bcos%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A15%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A35%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/clean_cache.sh%3AL1-47%22%7D%2C%20%22Pull%203ec01471b0598b5af99fa69e1767f979db7837aa%20tools/build.sh%2051%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211841725%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20r%20you%20omit%20backing%20to%20current_dir%3F%22%2C%20%22created_at%22%3A%20%222018-08-22T06%3A32%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/23206367%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wheatli%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A04%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/23206367%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wheatli%22%7D%7D%2C%20%7B%22body%22%3A%20%22already%20add%20%60cd%20%24%7Bcurrent_dir%7D%60%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A32%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A35%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/build.sh%3AL135-142%22%7D%2C%20%22Pull%203ec01471b0598b5af99fa69e1767f979db7837aa%20tools/build.sh%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211840998%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22pls%20check%20the%20url%2C%20maybe%20it%20is%20deprecated%22%2C%20%22created_at%22%3A%20%222018-08-22T06%3A28%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/23206367%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wheatli%22%7D%7D%2C%20%7B%22body%22%3A%20%22the%20url%20is%20in%20use%20now%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A13%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A35%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/build.sh%3AL1-13%22%7D%2C%20%22Pull%203ec01471b0598b5af99fa69e1767f979db7837aa%20tools/clean_cache.sh%203%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211841905%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22change%20to%20your%20own%20name%20pls%22%2C%20%22created_at%22%3A%20%222018-08-22T06%3A33%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/23206367%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wheatli%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A15%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A35%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/clean_cache.sh%3AL1-47%22%7D%2C%20%22Pull%203ec01471b0598b5af99fa69e1767f979db7837aa%20tools/build.sh%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211841464%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22use%20your%20name%20pls%22%2C%20%22created_at%22%3A%20%222018-08-22T06%3A31%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/23206367%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wheatli%22%7D%7D%2C%20%7B%22body%22%3A%20%22already%20updated%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A14%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222018-08-22T08%3A35%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/4343461%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyjseagull%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/build.sh%3AL1-13%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23issuecomment-414948040%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23issuecomment-414966382%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211840998%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211841464%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211841725%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211841827%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211841905%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211862590%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211864949%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211865412%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211865531%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211865595%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211870933%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211871661%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211871667%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211871669%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211871674%22%2C%20%22https%3A//github.com/FISCO-BCOS/lab-bcos/pull/4%23discussion_r211871678%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/wheatli'></a>

- [x] <a href='#crh-comment-Pull 3ec01471b0598b5af99fa69e1767f979db7837aa tools/build.sh 7'></a> &nbsp;<b><a href='https://github.com/FISCO-BCOS/lab-bcos/pull/4#discussion_r211840998'>File: tools/build.sh:L1-13</a></b>
- <a href='https://github.com/wheatli'></a> pls check the url, maybe it is deprecated
- <a href='https://github.com/cyjseagull'></a> the url is in use now
- [x] <a href='#crh-comment-Pull 3ec01471b0598b5af99fa69e1767f979db7837aa tools/build.sh 12'></a> &nbsp;<b><a href='https://github.com/FISCO-BCOS/lab-bcos/pull/4#discussion_r211841464'>File: tools/build.sh:L1-13</a></b>
- <a href='https://github.com/wheatli'></a> use your name pls
- <a href='https://github.com/cyjseagull'></a> already updated
- [x] <a href='#crh-comment-Pull 3ec01471b0598b5af99fa69e1767f979db7837aa tools/build.sh 51'></a> &nbsp;<b><a href='https://github.com/FISCO-BCOS/lab-bcos/pull/4#discussion_r211841725'>File: tools/build.sh:L135-142</a></b>
- <a href='https://github.com/wheatli'></a> why r you omit backing to current_dir?
- <a href='https://github.com/wheatli'></a> :+1:
- <a href='https://github.com/cyjseagull'></a> already add `cd ${current_dir}`
- [x] <a href='#crh-comment-Pull 3ec01471b0598b5af99fa69e1767f979db7837aa tools/clean_cache.sh 1'></a> &nbsp;<b><a href='https://github.com/FISCO-BCOS/lab-bcos/pull/4#discussion_r211841827'>File: tools/clean_cache.sh:L1-47</a></b>
- <a href='https://github.com/wheatli'></a> owner should be fisco-bcos
- <a href='https://github.com/cyjseagull'></a> owner is modified to fisco-bcos
- [x] <a href='#crh-comment-Pull 3ec01471b0598b5af99fa69e1767f979db7837aa tools/clean_cache.sh 3'></a> &nbsp;<b><a href='https://github.com/FISCO-BCOS/lab-bcos/pull/4#discussion_r211841905'>File: tools/clean_cache.sh:L1-47</a></b>
- <a href='https://github.com/wheatli'></a> change to your own name pls
- <a href='https://github.com/cyjseagull'></a> done

<a href='https://www.codereviewhub.com/FISCO-BCOS/lab-bcos/pull/4?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/FISCO-BCOS/lab-bcos/pull/4?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/FISCO-BCOS/lab-bcos/pull/4?approve=0'></a>&nbsp;<a href='https://github.com/FISCO-BCOS/lab-bcos/pull/4'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: tools/build.sh [hunk 1]
```diff
@@ -1,12 +1,12 @@
  # "Copyright [2018] <fisco-dev>"
- # @ function on-click install shell script(appliable for centos and ubuntu)
- # @ Require: yum or apt, git are ready
+ # @ function : one-click install shell script(appliable for centos and ubuntu)
+ # @ Require  : yum or apt, git are ready
  # @ attention: if dependecies are downloaded failed, 
  #              please fetch packages into "deps/src" from https://github.com/bcosorg/lib manually
```
Comment (wheatli) [inline tools/build.sh]: pls check the url, maybe it is deprecated

File: tools/build.sh [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
  #              and execute this shell script later
- # @ author: fisco-dev  
- # @ file: build.sh
- # @ date: 2018
+ # @ author   : fisco-dev
```
Comment (wheatli) [inline tools/build.sh]: use your name pls

File: tools/build.sh [hunk 3]
```diff
@@ -134,7 +135,7 @@ build_ubuntu_source() {
     execute_cmd "mkdir -p build && cd build/"
     execute_cmd "cmake .. "
     execute_cmd "make"
-    #execute_cmd "make install  && cd ${current_dir}"
+    #execute_cmd "make install"
```
Comment (wheatli) [inline tools/build.sh]: why r you omit backing to current_dir?

File: tools/clean_cache.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
+ # "Copyright [2018] <fisco-dev>"
```
Comment (wheatli) [inline tools/clean_cache.sh]: owner should be fisco-bcos

File: tools/clean_cache.sh [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ # @ function: clean cache
+ # @ author  : fisco-dev
```
Comment (wheatli) [inline tools/clean_cache.sh]: change to your own name pls

**Comment by wheatli** [comment]
:shipit:

File: tools/build.sh (hunk 1 above)
Comment (cyjseagull) [reply tools/build.sh]: the url is in use now

File: tools/build.sh (hunk 2 above)
Comment (cyjseagull) [reply tools/build.sh]: already updated

File: tools/clean_cache.sh (hunk 4 above)
Comment (cyjseagull) [reply tools/clean_cache.sh]: owner is modified to fisco-bcos

File: tools/clean_cache.sh (hunk 5 above)
Comment (cyjseagull) [reply tools/clean_cache.sh]: done

File: tools/build.sh (hunk 3 above)
Comment (cyjseagull) [reply tools/build.sh]: already add `cd ${current_dir}`

File: tools/build.sh (hunk 1 above)
Comment (cyjseagull) [inline tools/build.sh]: :+1:<a href='#crh-update-none'></a>

File: tools/build.sh (hunk 2 above)
Comment (cyjseagull) [inline tools/build.sh]: :+1:<a href='#crh-update-none'></a>

File: tools/build.sh (hunk 3 above)
Comment (cyjseagull) [inline tools/build.sh]: :+1:<a href='#crh-update-complete'></a>

**Comment by wheatli** [comment]
:shipit:

**Pull request merged**